cnd/src/db/tables.rs

```diff
@@ -399,8 +399,15 @@ impl Sqlite {
                     local_swap_id
                 )
             })?;
-
-        Ok(())
+        // There should be exactly one row updated, everything else is an error
+        if updated_rows == 1 {
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!(
+                "Could not update halight refund identity for swap {}",
+                local_swap_id
+            ))
+        }
```

Comment:
Personally, I mildly prefer handling error cases separately:

```suggestion
        // There should be exactly one row updated, everything else is an error
        if updated_rows != 1 {
            Err(anyhow::anyhow!(
                "Could not update halight refund identity for swap {}",
                local_swap_id
            ))
        }
        
        Ok(())
```

This way, the success path is visually different from the error path :)

Reply:
Completely agree that the _return early_ best practice is way nicer.